config/config.go

```diff
@@ -244,12 +245,19 @@ func loadExtensions(v *viper.Viper, factories map[string]extension.Factory) (con
 		extensionCfg.SetType(typeStr)
 		extensionCfg.SetName(fullName)
 
-		// Now that the default config struct is created we can Unmarshal into it
-		// and it will apply user-defined config on top of the default.
-		if err := subViper.UnmarshalKey(key, extensionCfg); err != nil {
-			return nil, &configError{
-				code: errUnmarshalError,
-				msg:  fmt.Sprintf("error reading settings for extension type %q: %v", typeStr, err),
+		// Unmarsh only the subconfig for this extension.
+		sv := subViper.Sub(key)
```

Comment:
These 4 lines are repeated 4 times (for extensions, exporters, receivers,processors). Perhaps make a func like `getConfigSection` which gets the sub key and expands env vars.

Reply:
Done.